### DISCUSSION TITLE
Add telemetry information for instrumentation/coverage calculation

### DISCUSSION THREAD

**Discussion by MarcoRossignoli** [coverlet-coverage/coverlet#1249] [Ideas]
We should add some telemetry instrumentation logging to be able to understand from logs where we have perf issue.
Interesting data would be:
* size of asm in MB
* time spent in main phases
* hits count per assembly
* ...

We can emit by design or add a flag like `--/<> CoverletTelemetry` to avoid to pollute too much logs for big projects.

Ref issue https://github.com/coverlet-coverage/coverlet/issues/1246#issuecomment-962494116

cc: @daveMueller @petli @tonerdo

**Comment by Bertk** [comment]
Closing this discussion since it’s no longer relevant to the current state of the project.
